### PR TITLE
Create a Demo dpdk build and deployment

### PR DESCRIPTION
**Pull request by SchSeba**
This PR creates a BuildConfig and a DeploymentConfig for a dpdk application.

I also moved the dpdk network from the sriov to the DPDK folder.

Signed-off-by: Sebastian Sch <sebassch@gmail.com>

### CODE REVIEW

**Comment by SchSeba** [comment]
/hold

I need to test this

File: feature-configs/demo/dpdk/build-config.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  name: s2i-dpdk-app
+  namespace: dpdk-testing
+spec: {}
+---
+apiVersion: build.openshift.io/v1
+kind: BuildConfig
```
Comment (davidvossel) [inline feature-configs/demo/dpdk/build-config.yaml]: after deploying these configs, is there a "wait" step that we can introduce that verifies the build is successful 

look at the `make feature-wait` logic

Comment (SchSeba) [reply feature-configs/demo/dpdk/build-config.yaml]: Hi @davidvossel thanks for the comment can you take another look?

File: feature-configs/demo/dpdk/is_ready.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+oc -n dpdk-testing get build -l app=s2i-dpdk
+
+oc -n dpdk-testing wait build --all --for condition=Complete --timeout 1s
+
+oc get dc s2i-dpdk-app
```
Comment (fedepaol) [inline feature-configs/demo/dpdk/is_ready.sh]: Why we need this?

File: feature-configs/demo/dpdk/is_ready.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+oc -n dpdk-testing get build -l app=s2i-dpdk
+
+oc -n dpdk-testing wait build --all --for condition=Complete --timeout 1s
```
Comment (fedepaol) [inline feature-configs/demo/dpdk/is_ready.sh]: would it make sense to wait for the only dc we care of (`s2i-dpdk`)?

File: feature-configs/demo/dpdk/build-config.yaml (hunk 1 above)
Comment (fedepaol) [reply feature-configs/demo/dpdk/build-config.yaml]: I think this is being covered now.

File: feature-configs/demo/dpdk/is_ready.sh (hunk 3 above)
Comment (SchSeba) [reply feature-configs/demo/dpdk/is_ready.sh]: The buildConfig doesn't have a condition field this is why I need to look onto the build and the build have a running number this is why I just look on all of them.

File: feature-configs/demo/dpdk/is_ready.sh (hunk 2 above)
Comment (SchSeba) [reply feature-configs/demo/dpdk/is_ready.sh]: To validate there is a deploymentConfig (if there is not it will return an exit code -1)

**Comment by SchSeba** [comment]
@fedepaol yes you are right the dpdk feature need sriov do you want to include it?

**Comment by fedepaol** [comment]
> @fedepaol yes you are right the dpdk feature need sriov do you want to include it?

Nope I was just thinking out loudly. If we don't deploy `sriov` the deployment will fail as the relevant network CRDs won't be there anyway. Having it in a loop has the nice effect that the order does not matter.

Maybe just adding a comment on top, wdyt?

**Comment by fedepaol** [comment]
/override ci/prow/e2e-gcp-origin

We set it as optional in https://github.com/openshift/release/pull/7005

**Comment by SchSeba** [comment]
@fedepaol can you run the override again? I don't have permissions for that

**Comment by fedepaol** [comment]
> @fedepaol can you run the override again? I don't have permissions for that

There's no need anymore, as we fixed the issue of having the origin tests mandatory

**Comment by SchSeba** [comment]
@fedepaol can you add the lgtm label again? :)

**Pull request merged**